beginner_source/torchtext_custom_dataset_tutorial.py

```diff
@@ -0,0 +1,383 @@
+
+# %%
+# The code above, builds the vocabulary from the iterator. In the above code block:
+#
+# * At line 2, we call the `get_tokens()` function with `place=0` as we need vocabulary for
+#   source sentences.
+# * At line 3, we set `min_freq=2`. This means, the function will skip those words that occurs
+#   less than 2 times.
+# * At line 4, we specify some special tokens:
+#
+#   * `<sos>` for start of sentence
+#   * `<eos>` for end of senetence
+#   * `<unk>` for unknown words. An example of unknown word is the one skipped because of
+#     `min_freq=2`.
+#   * `<pad>` is the padding token. While training, a model we mostly train in batches. In a
+#     batch, there can be sentences of different length. So, we pad the shorter sentences with
+#     `<pad>` token to make length of all sequences in the batch equal.
+#
+# * At line 5, we set `special_first=True`. Which means `<pad>` will get index 0, `<sos>` index 1,
+#   `<eos>` index 2, and <unk> will get index 3 in the vocabulary.
+# * At line 7, we set default index as index of `<unk>`. That means if some word is not in
+#   vocbulary, we will use `<unk>` instead of that unknown word.
+#
+# Similarly, we will build vocabulary for target sentences:
+
+targetVocab = build_vocab_from_iterator(
+    get_tokens(dataPipe,1),
+    min_freq=2,
+    specials= ['<pad>', '<sos>', '<eos>', '<unk>'],
+    special_first=True
+)
+targetVocab.set_default_index(targetVocab['<unk>'])
+
+# %%
+# Note that the example above shows how can we add special tokens to our vocabulary. The
+# special tokens may change based on the requirements.
+#
+# Now, we can verify that special tokens are placed at the beginning and then other words.
+# In the below code, `sourceVocab.get_itos()` returns a list with tokens at index based on
+# vocabulary.
+
+print(sourceVocab.get_itos()[:9])
+
+# %%
+# Numericalize sentences using vocabulary
+# ---------------------------------------
+# After building the vocabulary, we need to convert our sentences to corresponding indices.
+# Let us define some functions for this:
+
+def get_transform(vocab):
+    """
+    Create transforms based on given vocabulary. The returned transform is applied to sequence
+    of tokens.
+    """
+    text_tranform = T.Sequential(
+        ## converts the sentences to indices based on given vocabulary
+        T.VocabTransform(vocab=vocab),
+        ## Add <sos> at beginning of each sentence. 1 because the index for <sos> in vocabulary is
+        # 1 as seen in previous section
+        T.AddToken(1, begin=True),
+        ## Add <eos> at beginning of each sentence. 2 because the index for <eos> in vocabulary is
+        # 2 as seen in previous section
+        T.AddToken(2, begin=False)
+    )
+    return text_tranform
+
+# %%
+# Now, let us see how to use the above function. The function returns an object of `Transforms`
+# which we will use on our sentence. Let us take a random sentence and check the working of
+# the transform:
```

Comment:
nit: rephrase to "and check how the transform works."

Reply:
I have made changes as suggested.